### PR TITLE
Adds basic visBuilder test

### PR DESCRIPTION
**Pull request by ashwin-pc**
Signed-off-by: Ashwin Pc <ashwinpc@amazon.com>

### Description

- Adds basic visBuilder test
- Adds common custom commands:
  - bulkUploadDocs
  - createIndexPattern
  - deleteIndexPattern
  - drag (for drag and drop)
- Adds Dashboards specific commands:
  - setTopNavQuery
  - setTopNavDate
  - updateTopNav
- Adds Visbuilder commands:
  - vbSelectDataSource
  - vbSelectVisType
- Adds type interface for custom commands to expose them to the IDE

https://user-images.githubusercontent.com/20453492/194674153-6d04ecff-b841-4d2c-899b-0bd1f6c3031c.mp4

Test command: 

```sh
CYPRESS_VISBUILDER_ENABLED=true  yarn  cypress:run-without-security --spec 'cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/**/*.js'
```
 
### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tianleh** [comment]
Overall looks good. Could you add the test command and the test video output to the issue for trackability?

**Comment by ashwin-pc** [comment]
@tianleh Thanks :) I can add the test command but i see that the "Bundle Snapshot based E2E Cypress tests workflow for core Dashboards / Run Cypress E2E te.." workflow fails. Do you know why thats the case? Thats the test responsible for running core dashboards tests. right now its failing to even startup dashboards

**Comment by tianleh** [comment]
> @tianleh Thanks :) I can add the test command but i see that the "Bundle Snapshot based E2E Cypress tests workflow for core Dashboards / Run Cypress E2E te.." workflow fails. Do you know why thats the case? Thats the test responsible for running core dashboards tests. right now its failing to even startup dashboards

Security team is fixing it. https://github.com/opensearch-project/opensearch-build/issues/2708

**Comment by tianleh** [comment]
It may take a hour or two to complete the new build and the update of `latest` url at infra side. You can then rerun the workflow. I am flexible to ship if you have completed the local testing and attach the command and video here. Just let me know which way you want to proceed. @ashwin-pc

File: .github/workflows/cypress-workflow-bundle-snapshot-based.yml
Comment (ashwin-pc) [inline .github/workflows/cypress-workflow-bundle-snapshot-based.yml:28]: I moved the bundle snapshot workflow to use the template since they were identical. The only difference was that the bundle workflow user `monetery-test` instead of `cypress-test`. I assume that this change is safe.Additionally the template also allows us to skip tests if there are no changes to the spec files that it works on, speeding up the CI

**Comment by ashwin-pc** [comment]
I'm okay with either, but i'd love to ideally see that my changes didnt break anything since I have moved the snapshot workflow to now use the template

File: .github/workflows/release-e2e-workflow-template.yml
Comment (tianleh) [inline .github/workflows/release-e2e-workflow-template.yml:11]: Elegant solution!

File: .github/workflows/cypress-workflow-bundle-snapshot-based.yml
Comment (tianleh) [reply .github/workflows/cypress-workflow-bundle-snapshot-based.yml:28]: Thanks for the refactoring!

**Comment by tianleh** [comment]
> I'm okay with either, but i'd love to ideally see that my changes didnt break anything since I have moved the snapshot workflow to now use the template

Make sense! Probably then let's wait for the `latest` 3.0.0 to work and rerun the workflow as a final check before merging the PR. (by the way, the video looks great. better to share your exact test command as well.)

**Review by tianleh — APPROVED** [review]

**Review by seraphjiang — APPROVED** [review]

**Comment by tianleh** [comment]
@ashwin-pc Does this need to backport to 2.x/2.4? Currently it is merged to `main` which is for `3.0.0`

**Comment by ashwin-pc** [comment]
@tianleh yes we need this to be back ported to 2.4 and 2.x. Thanks for calling it out :)

**Pull request merged**
